Fixed three eslint issues -

manurampandit opened the pull request:
As part of https://github.com/ember-learn/ember-website/issues/672 , fixing three other linting issues.
what-wg-fetch addition is due to test failing. Reference taken from - https://github.com/ember-cli/ember-fetch/issues/547

Have tested this in browser and running 'ember t' 

@ijlee2 - Kindly review this PR as well.

mansona commented on package.json:
```diff
@@ -88,6 +88,9 @@
       "lib/content-data-generator"
     ]
   },
+  "resolutions": {
```
so this won't work for a number of reasons: 

- this project is not using yarn so resolutions will not have any effect
- we should not be relying on resolutions to fix issues like this
- it seems like this problem will cause issues after upgrading as mentioned in https://github.com/ember-cli/ember-fetch/issues/547#issuecomment-671885942

can you revert this change, and if it is still causing an issue then we can look at upgrading ember-fetch which seems to have fixed the issue 👍

manurampandit replied:
Thanks a lot for reviewing this PR. I've updated ember-fetch version as this issue was still occurring.

mansona commented on app/routes/application.js:
```diff
@@ -25,7 +25,7 @@ export default Route.extend({
 
     const page = this.url;
-    const title = this.getWithDefault('currentRouteName', 'unknown');
+    const title = this.currentRouteName || 'unknown';
```
can you please make use of `??` instead of `||`. I know in this case the currentRouteName is supposed to be a string so it will only be falsey if it is empty, but I would prefer to make use of `??` for default values as a standard 👍

manurampandit replied:
Thank you for reminding me[ Coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) use. Almost forgot :)

mansona commented on app/controllers/community/meetups/index.js:
```diff
@@ -10,7 +10,7 @@ export default Controller.extend({
     this._super(...arguments);
     
-    if (!this.get('fastboot.isFastBoot')) {
+    if (!this.fastboot.isFastBoot) {
```
this has actually subtly changed the semantics of the line. Now if fastboot is null or undefined it will throw an error (where it would not have before).

For us to keep the same semantics you would need to do something like this: 

```suggestion
    if (!this.fastboot?.isFastBoot) {
```

manurampandit replied:
Updated this.

mansona commented on app/components/high-charts.js:
```diff
@@ -161,9 +161,9 @@ export default Component.extend({
   draw() {
     let element = this.element && this.element.querySelector('.chart-container');
-    let modeAttr = get(this, 'mode') || undefined;
+    let modeAttr = this.mode || undefined;
```
```suggestion
    let modeAttr = this.mode ?? undefined;
```

mansona commented on app/components/high-charts.js:
```diff
@@ -79,14 +79,14 @@ export default Component.extend({
 
     // if 'no-data-to-display' module has been imported, keep empty series and leave it to highcharts to show no data label.
-    if (!get(this, 'content.length') && !this.highcharts.Chart.prototype.showNoData) {
+    if (!this.content.length && !this.highcharts.Chart.prototype.showNoData) {
```
again this breaks the semantics of what it was doing before: 

This will fix it: 
```suggestion
    if (!this.content?.length && !this.highcharts.Chart.prototype.showNoData) {
```

mansona commented on app/components/high-charts.js:
```diff
@@ -79,14 +79,14 @@ export default Component.extend({
-    let theme = getWithDefault(this, 'theme', {});
-    let passedChartOptions = getWithDefault(this, 'chartOptions', {});
+    let theme = this.theme || {};
+    let passedChartOptions = this.chartOptions || {};
```
```suggestion
    let passedChartOptions = this.chartOptions ?? {};
```

mansona commented on app/components/high-charts.js:
```diff
@@ -79,14 +79,14 @@ export default Component.extend({
   buildOptions: computed('chartOptions', 'content.[]', function() {
-    let theme = getWithDefault(this, 'theme', {});
-    let passedChartOptions = getWithDefault(this, 'chartOptions', {});
+    let theme = this.theme || {};
```
```suggestion
    let theme = this.theme ?? {};
```

mansona reviewed (changes requested):
Hey thanks for your contribution 🎉 I've added a few comments that will probably need to be looked at before we can get this merged, but overall it's a great job 👍

manurampandit commented:
Thank you @mansona for reviewing this and providing qualitative feedback. I've addressed all the comments. Kindly re-review and let me know if I need to change anything else.
